From 151c00e4c9f85a00d84f3a2ba9e4751c15915e4c Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 31 Aug 2015 11:37:11 +0800
Subject: [PATCH] MLK-11456-02 ARM: dts: imx7d-12x12-lpddr3-arm2: enable uart3
 for bt in lpsr mode

Enable uart3 for bt in lpsr mode.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index 3f8c9e7..1cb6f34 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -481,7 +481,6 @@
 				MX7D_PAD_ECSPI2_SCLK__GPIO4_IO20  0x80000000  /* v3p3 */
 				MX7D_PAD_ECSPI2_MOSI__GPIO4_IO21  0x80000000  /* pwrctrl1 */
 				MX7D_PAD_ECSPI2_MISO__GPIO4_IO22  0x80000000  /* pwrctrl2 */
-				MX7D_PAD_ENET1_CRS__GPIO7_IO14    0x80000000  /* BT REG on */
 				MX7D_PAD_SD1_RESET_B__GPIO5_IO2   0x59
 				MX7D_PAD_SD1_CD_B__GPIO5_IO0      0x59
 				MX7D_PAD_SD1_WP__GPIO5_IO1        0x59
@@ -504,6 +503,12 @@
 			>;
 		};
 
+		pinctrl_bt: btgrp-1 {
+			fsl,pins = <
+				MX7D_PAD_ENET1_CRS__GPIO7_IO14    0x80000000  /* BT REG on */
+			>;
+		};
+
 		pinctrl_ecspi1_cs_1: ecspi1_cs_grp-1 {
 			fsl,pins = <
 				MX7D_PAD_ECSPI1_SS0__GPIO4_IO19     0x2
@@ -955,20 +960,24 @@
 };
 
 &uart1 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_uart1_1>;
+	pinctrl-1 = <&pinctrl_uart1_1>;
 	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
 	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
 	status = "okay";
 };
 
 &uart3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart3_1>;
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&pinctrl_uart3_1
+		     &pinctrl_bt>;
+	pinctrl-1 = <&pinctrl_uart3_1
+		     &pinctrl_bt>;
 	fsl,uart-has-rtscts;
 	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
 	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
-	status = "disabled";
+	status = "okay";
 	/* for DTE mode, add below change */
 	/* fsl,dte-mode;*/
 	/* pinctrl-0 = <&pinctrl_uart3dte_1>; */
-- 
1.8.0

